zebra-chain/src/sapling/tree.rs

```diff
@@ -498,10 +487,11 @@ impl From<&NoteCommitmentTree> for SerializedTree {
         //
         // [1]: <https://github.com/zcash/librustzcash/blob/a63a37a/zcash_primitives/src/merkle_tree.rs#L125>
         if let Some(frontier) = tree.inner.value() {
-            let (left_leaf, right_leaf) = match frontier.leaf() {
-                Leaf::Left(left_value) => (Some(left_value), None),
-                Leaf::Right(left_value, right_value) => (Some(left_value), Some(right_value)),
-            };
```

Comment:
We can't do any of these anymore. I think we will just have to go with `write_frontier_v1` just as orchard do and delete the rest of the code. I am unsure if the doc comments above make sense here or in orchard so we might want to remove them too.

We could also remove `SerializedTree` fully but i don't think we should do that in this PR.

Reply:
The JSON-RPC format used by `zcashd` is `zcash_primitives::merkle_tree::write_commitment_tree()`, so I think we have to use that:
https://docs.rs/zcash_primitives/0.12.0/zcash_primitives/merkle_tree/fn.write_commitment_tree.html

I don't know why we implemented the serialization ourselves, maybe that method didn't exist, or we didn't know about it.

Here's how I found that:

Orchard tree RPC:
https://github.com/zcash/zcash/blob/4c32e3b3e3e0f576cf0c37aa260a5a9d1619ec3b/src/rpc/blockchain.cpp#L1432

OrchardMerkleFrontierLegacySer type:
https://github.com/zcash/zcash/blob/4c32e3b3e3e0f576cf0c37aa260a5a9d1619ec3b/src/zcash/IncrementalMerkleTree.hpp#L423C29-L423C45

serialize_legacy method:
https://github.com/zcash/zcash/blob/4c32e3b3e3e0f576cf0c37aa260a5a9d1619ec3b/src/rust/src/merkle_frontier.rs#L47

write_commitment_tree_method:
https://github.com/zcash/librustzcash/blob/4d5dc28ab18a020f46a87c1b76d0bf7883ae55c7/zcash_primitives/src/merkle_tree.rs#L182

Reply:
In zcash_primitives 0.11 `write_commitment_tree()` is `CommitmentTree::write()`:
https://docs.rs/zcash_primitives/0.11.0/zcash_primitives/merkle_tree/struct.CommitmentTree.html#method.write

I can make this change in a separate PR if it's easier?

Reply:
I think we have to do the upgrade first, then use `CommitmentTree::from_parts()` to replace our custom Sapling code:
https://docs.rs/incrementalmerkletree/0.4.0/src/incrementalmerkletree/frontier.rs.html#321

Reply:
This code was converted to what it was by converting the `NoteCommitmentTree` into a `LegacyNoteCommitmentTree` in https://github.com/ZcashFoundation/zebra/pull/7053/commits/f07c5716e52c7e47e21651971a9791512a76802f

Also for orchard the old code was added back (always with some minor mods to make it work) in https://github.com/ZcashFoundation/zebra/pull/7053/commits/7962f315163fea30d7fef26c62e91454ce44b62d

I don't think we should try to use new functions or way of doing it better in this PR and just stick with what we know it was working.